src/BCEngineMCMC.cxx

```diff
@@ -1239,40 +1230,40 @@ void BCEngineMCMC::Remarginalize(bool autorange)
 
     // link log(probability) if available
     if (fMCMCTree->GetBranch("LogProbability"))
-        fMCMCTree->SetBranchAddress("LogProbability", &fMCMCTree_Prob);
+        fMCMCTree->SetBranchAddress("LogProbability", &fMCMCTree_State.log_probability);
 
     // link iteration if available
-    bool has_iteration = true;
     if (fMCMCTree->GetBranch("Iteration"))
-        fMCMCTree->SetBranchAddress("Iteration",      &fMCMCTree_Iteration);
+        fMCMCTree->SetBranchAddress("Iteration",      &fMCMCTree_State.iteration);
     else
-        has_iteration = false;
+        fMCMCTree_State.iteration = 0;
 
     // link log(likelihood) if available
     if (fMCMCTree->GetBranch("LogLikelihood"))
```

Comment:
Why is and was likelihood used here? I thought we had [`BCModel::InitializeMarkovChainTree` for that. We shouldn't have any variable named  `likelihood` inside the engine

Reply:
I don't understand your comment in reference to the highlighted code.

This function is meant to load in a `TTree` and fill all possible marginalization information given the information in the tree. It's structured to not presume the tree has been made by BAT. At it's minimum, I believe it only requires parameters be in the tree.